pkg/service/configcache/service_test.go

```diff
@@ -0,0 +1,232 @@
+// Copyright (C) 2024 ScyllaDB
+
+package configcache
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/json"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/scylladb/scylla-manager/v3/pkg/scyllaclient"
+	"github.com/scylladb/scylla-manager/v3/pkg/service/cluster"
+	"github.com/scylladb/scylla-manager/v3/pkg/store"
+	"github.com/scylladb/scylla-manager/v3/pkg/util/uuid"
+)
+
+// Servicer.ForceUpdateCluster() and Servicer.Init(ctx context.Context) are expected to be covered with the
+// integration tests.
+
+func TestService_Read(t *testing.T) {
+	emptyConfigHash, err := NodeConfig{}.sha256hash()
+	if err != nil {
+		t.Fatalf("unable to create sha256 hash out of empty NodeConfig, err = {%v}", err)
+	}
+	host1NodeConfig := NodeConfig{
+		NodeInfo: &scyllaclient.NodeInfo{
+			AgentVersion: "expectedVersion",
+		},
+	}
+	host1ConfigHash, err := host1NodeConfig.sha256hash()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	cluster1UUID, err := uuid.NewRandom()
+	host1ID := "host1"
+	if err != nil {
+		t.Fatalf("unable to generate random UUID for test purpose")
+	}
+	initialState := convertMapToSyncMap(
+		map[any]any{
+			cluster1UUID.String(): convertMapToSyncMap(
+				map[any]any{
+					host1ID: host1NodeConfig,
+				},
+			),
+		},
+	)
+
+	for _, tc := range []struct {
+		name             string
+		cluster          uuid.UUID
+		host             string
+		state            *sync.Map
+		resultErr        error
+		resultConfigHash [32]byte
```

Comment:
What wrong do you see in comparing hashes ? Hash collision ?

Reply:
Hash collisions are not a problem, but having clear description of result config instead of just its hash makes the tests more comprehensible. Comparing by struct with `cmp.Diff` also allows for detecting which exact fields are different and it does not require to define a method of hashing `NodeConfig` struct.